src/java/org/httpkit/client/HttpClient.java

```diff
@@ -309,8 +310,10 @@ public void exec(String url, RequestConfig cfg, SSLEngine engine, IRespListener
         try {
             if (proxyUri == null) {
                 addr = addressFinder.findAddress(uri);
+                host = uri.getHost();
             } else {
                 addr = addressFinder.findAddress(proxyUri);
+                host = proxyUri.getHost();
```

Comment:
In this case, I think that we need to use something like `proxyUri.getHost() + uri.getHost()`